### ISSUE TITLE
Make all internal URLs ~relative~ absolute-path paths

### ISSUE THREAD

**Issue by philwareham** [textpattern/textpattern.github.io#114]
So people can build and test the Jekyll site locally, we should make all internal URLs ~relative~ absolute-path paths and not enforce domain name.

**Comment by petecooper** [comment]
…so `https://docs.textpattern.com/administration/` becomes `/administration/`?

**Comment by philwareham** [comment]
Hi Pete - yeah, I've just committed some that I had done yesterday. Not had a chance to continue with it yet. Its not a simple search and remove domain though, takes some time to sift through all the URLs and change.

**Comment by Bloke** [comment]
Can't we just knock the domain and prefix on the head? Is there any reason, for example, in the [user roles](https://github.com/textpattern/textpattern.github.io/commit/0a72c74835c70e47da8eea7e7a2e0ca46bf8e883#diff-e7ebba3aab6b06db98e103ad03a6ec45R11) link not to use:

`/administration/user-roles-and-privileges` ?

Browsers only automatically add the current section if the URL is not root-relative. As long as GitHub-Flavoured Markdown uses what we give it and permits the same, then it _could_ become a simple case of search and replace, right?

**Comment by philwareham** [comment]
Yeah, that would work fine if using a local web server instance (which I guess local testers/users would. I was just thinking ahead to if we try to turn this into a downloadable document (where I think relative URLs might be more robust). Let me know your thoughts?

**Comment by petecooper** [comment]
>I was just thinking ahead to if we try to turn this into a downloadable document (where I think relative URLs might be more robust). Let me know your thoughts?

I've done this before, Pandoc would handle this without issue.

**Comment by Bloke** [comment]
Hmm, depends what you mean by 'downloadable document'. If we're going to try and convert the entire site into a single PDF then goodness knows what'll happen to internal links. I don't think anything will save us! Especially if we refer to something of the same name in two sections (can't think of anything offhand but I would expect there to be some).

If we're talking about making the docs available as a set of _text_ documents then links won't be honoured anyway, regardless of the mechanism used.

If we're making them a series of static HTML files then they'll need to be browsed in a browser anyway, so root-relative links will be fine; the browser will (ahem, should) handle prepending `file://` prefixes and navigation will be the same as on a web server. That will also preserve JavaScript, with any luck.

Creation of a single PDF for the tags might be nice, or creation of a PDF from a single section. But if that section links to other sections outside (which I'm sure we do in the tags, for example, when we reference the Prefs panel) then the link will "go nowhere" if we havebn't included that section in the doc.

In short, I think root-relative is probably what we want, but I'm willing to be proven wrong. A browsable static set of .html docs is most likely the minimum at we should aim for offline perusal situations.

**Comment by philwareham** [comment]
Yes we should do this :)

**Comment by Bloke** [comment]
Cool. Reckon we can use sed/grep to go through and do it en-masse in a local clone? Something like...

```
find . -type f -exec sed -i 's/https:\/\/docs.textpattern.com//g' {} +
```

:D

**Comment by Bloke** [comment]
Haha, it sorta works. Kind of busts a few pages we don't want to change, but it's close.

**Comment by Bloke** [comment]
On MacOS, this beast seems to do the trick:

```
find . \( ! -regex '.*/\..*' \) -type f -exec sed -i '' -e 's/https:\/\/docs.textpattern.com\//\//g' {} \;
```

It complains about `sed: RE error: illegal byte sequence` a lot, but seems to have replaced all the instances without ill effect to any of the hidden files. I can apply it to the repo if you like.

**Comment by philwareham** [comment]
I have just don't a find and replace in my IDE, selectively ignoring ones we don't want to change (like the `itemprop` values which should be fully resolved URLs). I will commit to repo soon.

**Comment by petecooper** [comment]
Hey - just saw this and my name attached - what can I do?

**Comment by Bloke** [comment]
Nice!

**Comment by philwareham** [comment]
I think I've done all this now. If you spot any errors please feel free to fix. :)

**Comment by wion** [comment]
Sorry, guys. I’m not following this well. Are we supposed to be using relative or absolute? I’ve always thought it was the latter because of some constraint, and then lost touch on the issue.

So, if I’m editing page A and want to link to page B, this is the path pattern to use?

~~~
[page B](/tags/tag-b)
~~~

**Comment by Bloke** [comment]
Yep!

**Comment by wion** [comment]
Same for image references, I guess?

**Comment by Bloke** [comment]
Aye. Any resource that's relative to the docs site shouldn't need the full domain. Only stuff that's off-site will need it.

Just remember, for local links, to use root-relative (starting with a slash) and *not* just relative links (e.g. using `../` is not good).

**Comment by Bloke** [comment]
It appears a side-effect of doing this is that, after editing docs on the GitHub repo, you can no longer test the doc links directly from there, because their root-relative nature adds the github.com domain as prefix.

So to see them in action, you need to wait for the CI build to go through and then refresh the public docs page. No biggie, just for awareness.
